src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java

```diff
@@ -0,0 +1,155 @@
+    private final Map<String, String> indexPathMap = new HashMap();
+
+    public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state) {
+        this.delegate = delegate;
+        this.state = state;
+        this.nativeMemoryCacheManager = NativeMemoryCacheManager.getInstance();
+
+        Directory directory = state.directory;
+        // directory would be CompoundDirectory, we need get directory firstly and then unwrap
+        if (state.directory instanceof KNN80CompoundDirectory) {
+            directory = ((KNN80CompoundDirectory) state.directory).getDir();
+        }
+        String directoryPath = ((FSDirectory) FilterDirectory.unwrap(directory)).getDirectory().toString();
+        for (FieldInfo field : state.fieldInfos) {
+            if (!field.attributes().containsKey(KNN_FIELD)) {
+                continue;
+            }
+            KNNEngine knnEngine = getKNNEngine(field);
+            List<String> engineFiles = getEngineFiles(knnEngine.getExtension(), field.name);
+            Path indexPath = PathUtils.get(directoryPath, engineFiles.get(0));
+            indexPathMap.putIfAbsent(field.getName(), indexPath.toString());
+        }
+    }
+
+    @Override
+    public BinaryDocValues getBinary(FieldInfo field) throws IOException {
+        return delegate.getBinary(field);
+    }
+
+    @Override
+    public NumericDocValues getNumeric(FieldInfo field) throws IOException {
+        return delegate.getNumeric(field);
+    }
+
+    @Override
+    public SortedDocValues getSorted(FieldInfo field) throws IOException {
+        return delegate.getSorted(field);
+    }
+
+    @Override
+    public SortedNumericDocValues getSortedNumeric(FieldInfo field) throws IOException {
+        return delegate.getSortedNumeric(field);
+    }
+
+    @Override
+    public SortedSetDocValues getSortedSet(FieldInfo field) throws IOException {
+        return delegate.getSortedSet(field);
+    }
+
+    @Override
+    public void checkIntegrity() throws IOException {
+        delegate.checkIntegrity();
+    }
+
+    @Override
+    public void close() {
+        try {
+            delegate.close();
+            for (String path : indexPathMap.values()) {
+                nativeMemoryCacheManager.invalidate(path);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public final List<String> getOpenedIndexPath() {
+        return new ArrayList<>(indexPathMap.values());
+    }
+
+    /**
+     * Get KNNEngine From FieldInfo
+     * @param field which field we need produce from engine
+     * @return if and only if Native Engine we return specific engine, else return Lucene Engine
+     */
+    private KNNEngine getKNNEngine(@NonNull FieldInfo field) {
+
+        final String modelId = field.attributes().get(MODEL_ID);
+        if (modelId != null) {
+            var model = ModelCache.getInstance().get(modelId);
+            return KNNEngine.LUCENE;
```

Comment:
Why Lucene here?

Reply:
nice catch, i just want to get nativeEngine file path. it is truly confused, so at https://github.com/opensearch-project/k-NN/blob/ee0b7d4217b0db76626973fe4209cc429a6147c3/src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java#L123-L144 i using specific KNNEngine enum or null